src/nvim/os/shell.c

```diff
@@ -405,22 +405,23 @@ static bool out_data_decide_throttle(size_t size)
     return false;
   } else if (!visit) {
     started = os_hrtime();
-  } else if (visit % 20 == 0) {
+  } else {
     uint64_t since = os_hrtime() - started;
     if (since > (3 * NS_1_SECOND)) {
       received = visit = 0;
       return false;
     }
+    if (since < (visit * 0.1 * NS_1_SECOND)) {
+      return true;
+    }
```

Comment:
`if (visit % 20 == 0)` was to avoid spamming `os_hrtime()`.

Reply:
..but we need it for rate limitting to 10/s.